File: solutions/Previous/221115/theLeastCommonMultiple.test.js
import solution, { greatestCommonDenominator, leastCommonMultiple } from './theLeastCommonMultiple';

test('최대공약수 계산', () => {
  expect(greatestCommonDenominator(12, 8)).toBe(4);
  expect(greatestCommonDenominator(2, 6)).toBe(2);
  expect(greatestCommonDenominator(8, 14)).toBe(2);
});

test('최소공배수 계산', () => {
  expect(leastCommonMultiple(12, 8)).toBe(24);
  expect(leastCommonMultiple(2, 6)).toBe(6);
  expect(leastCommonMultiple(8, 14)).toBe(56);
});

test('solution', () => {
  expect(solution([2, 6, 8, 14])).toBe(168);
  expect(solution([1, 2, 3])).toBe(6);
});

File: solutions/Previous/221114/TheLeastCommonMultipleTest.java
import org.junit.jupiter.api.Test;

import java.util.HashMap;
import java.util.Map;

import static org.assertj.core.api.Assertions.assertThat;

class TheLeastCommonMultipleTest {
//    @Test
//    void factorize() {
//        TheLeastCommonMultiple test = new TheLeastCommonMultiple();
//        Map<Integer, Integer> primeFactors = new HashMap<>();
//        primeFactors.put(2, 2);
//        primeFactors.put(3, 1);
//        primeFactors.put(5, 1);
//
//        assertThat(test.factorize(60)).isEqualTo(primeFactors);
//    }

    @Test
    void solution() {
        TheLeastCommonMultiple test = new TheLeastCommonMultiple();
        assertThat(test.solution(new int[]{2, 6, 8, 14}))
            .isEqualTo(168);
        assertThat(test.solution(new int[]{1, 2, 3}))
            .isEqualTo(6);
    }
}
